src/main/java/org/reaktivity/nukleus/http2/internal/Http2Stream.java

```diff
@@ -209,13 +205,10 @@ void onEnd()
         // more request data to be sent, so send ABORT
         if (state != Http2StreamState.HALF_CLOSED_REMOTE)
         {
-            httpWriteScheduler.doAbort(0);
+            httpWriteScheduler.doAbort(factory.supplyTrace.getAsLong());
         }
 
-        if (applicationReplyThrottle != null)
-        {
-            factory.doReset(applicationReplyThrottle, applicationRouteId, applicationReplyId, factory.supplyTrace.getAsLong());
-        }
+        factory.doReset(applicationInitial, applicationRouteId, applicationReplyId, factory.supplyTrace.getAsLong());
```

Comment:
Same feedback as above. Recommend moving the logic to a separate method avoid duplicate implementation.